### ISSUE TITLE
Route domain not working for bigip_virtual_server - ansible 2.5

### ISSUE THREAD

**Issue by jamiekowalczik** [F5Networks/f5-ansible#683]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

bigip_virtual_server
##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/project-ansible2.5/lib/python2.7/site-packages/ansible
  executable location = /home/user/project-ansible2.5/bin/ansible
  python version = 2.7.5 (default, Sep  5 2016, 02:30:38) [GCC 4.8.5 20150623 (Red Hat 4.8.5-9)]
```

##### PYTHON VERSION

```
Python 2.7.5
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     11.6.1
  Build       1.0.326
  Edition     Hotfix HF1
  Date        Fri Aug 19 16:22:57 PDT 2016
```

##### LIBRARY VERSIONS

```
bigsuds==1.0.6
f5-sdk==3.0.13
```

##### OS / ENVIRONMENT

RHEL 7.3
##### SUMMARY

Route domain is not used when adding the destination.  When using a route domain in the source address the following error is generated:
fatal: [bigip -> localhost]: FAILED! => {"changed": false, "msg": "The source IP address must be specified in CIDR format: address/prefix"}

##### STEPS TO REPRODUCE

```
---
- name: Create LTM Configuration
  hosts: bigip
  connection: local
  gather_facts: false

  tasks:
  - name: Create Virtual
    bigip_virtual_server:
       server: "{{ inventory_hostname }}"
       user: "{{ bigip_username }}"
       password: "{{ bigip_password }}"
       state: present
       name: "route_domain_bug-vs"
       description: "Route Domain Bug VS"
       profiles:
         - http
       partition: "{{ partition }}"
       source: 0.0.0.0%101/0
       destination: 1.1.1.10%101
       port: "443"
       validate_certs: false
    delegate_to: localhost
```

##### EXPECTED RESULTS

Using ansible 2.4.2 (using virtualenv to switch between versions) I am able to create the destination with the route domain and without source specified (not supported for that version) the route domain is added appropriately for source.

##### ACTUAL RESULTS

PLAY [Create LTM Configuration] *****************************************************************************************************************************************************************

TASK [Create Virtual] *****************************************************************************************************************************************************************************************
[DEPRECATION WARNING]: Param 'user' is deprecated. See the module docs for more information. This feature will be removed in version 2.9. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: Param 'password' is deprecated. See the module docs for more information. This feature will be removed in version 2.9. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: Param 'server' is deprecated. See the module docs for more information. This feature will be removed in version 2.9. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: Param 'validate_certs' is deprecated. See the module docs for more information. This feature will be removed in version 2.9. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
fatal: [bigip -> localhost]: FAILED! => {"changed": false, "msg": "The source IP address must be specified in CIDR format: address/prefix"}
        to retry, use: --limit @/playbooks/for_routedomain_bug.retry

PLAY RECAP ****************************************************************************************************************************************************************************************************
bigip    : ok=1    changed=0    unreachable=0    failed=1

```

```

**Comment by caphrim007** [comment]
@jamiekowalczik your description is a little unclear, is the route domain failing for `source`? `destination`? or both?

**Comment by jamiekowalczik** [comment]
Sorry @caphrim007. 

With the play above I get the error shown.  If I remove source the play finishes ok but the destination for the virtual is: 1.1.1.10 (without %101)

Using ansible 2.4.2 with the following play creates the virtual with the route domain (%101) added to both the source and destination as I would expect it but I was forced to upgrade to ansible 2.5 because of other challenges which now leaves me with this behavior.

---
- name: Create LTM Configuration
  hosts: bigip
  connection: local
  gather_facts: false

  tasks:
  - name: Create Virtual
    bigip_virtual_server:
       server: "{{ inventory_hostname }}"
       user: "{{ bigip_username }}"
       password: "{{ bigip_password }}"
       state: present
       name: "route_domain_bug-vs"
       description: "Route Domain Bug VS"
       all_profiles:
         - http
       partition: "{{ partition }}"
       destination: 1.1.1.10%101
       port: "443"
       validate_certs: false
    delegate_to: localhost

**Comment by caphrim007** [comment]
@jamiekowalczik are you sure it added it to the `source`? `source` was not an accepted parameter in 2.4

**Comment by jamiekowalczik** [comment]
Yes.  Without specifying source in my playbook in my ansible 2.4 environment it would add the route domain to the source as well as the destination.  I didn't previously use source in my plays until upgrading to 2.5 this morning.

(project-ansible2.4.2)[user@host project1]$ ansible --version
ansible 2.4.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/project-ansible2.4.2/lib/python2.7/site-packages/ansible
  executable location = /home/user/project-ansible2.4.2/bin/ansible
  python version = 2.7.5 (default, Sep  5 2016, 02:30:38) [GCC 4.8.5 20150623 (Red Hat 4.8.5-9)]

---
PLAY [Create LTM Configuration] *******************************************************************************************************************************************************************************

TASK [Create Virtual] *****************************************************************************************************************************************************************************************
changed: [bigip -> localhost]

PLAY RECAP ****************************************************************************************************************************************************************************************************
bigip    : ok=1    changed=1    unreachable=0    failed=0

-----
[root@bigip:Active:Standalone] ~ # tmsh list /ltm virtual /mypartition/route_domain_bug-vs
ltm virtual /mypartition/route_domain_bug-vs {
    description "Route Domain Bug VS"
    destination /Broome/1.1.1.10%101:https
    ip-protocol tcp
    mask 255.255.255.255
    partition mypartition
    profiles {
        http { }
        tcp { }
    }
    source 0.0.0.0%101/0
    vs-index 49
}

**Comment by caphrim007** [comment]
that seems to be a side-effect of mcpd then, as it was not an intended use of the virtual server module in 2.4. There were many such unintended situations with that module. The one in 2.5 is more intentional. Never-the-less, this is a bug and will be fixed so that it works once again.

**Comment by jamiekowalczik** [comment]
Much appreciated, thank you so much for your time and effort on this!

**Comment by wojtek0806** [comment]
this is now fixed

**Comment by prabhatkarki** [comment]
It appears the fix with 2.8 is only on IPv4. We are running on 2.8.3 and when using this module to create Virtual Server with route domain for IPv6 we have same issue that it ignores what comes after the %. 
"2001:4888:200:10:0:103:0:1%2"  gets added as 2001:4888:200:10:0:103:0:1
t works for ipv4. So is there a certain format that we need to use for IPv6. or is this a bug and needs to be fix within the module?
